frontend/resourceadm/pages/DeployResourcePage/DeployResourcePage.tsx

```diff
@@ -1,81 +1,153 @@
+        console.log(res);
+        const versions: ResourceVersionStatusType = res as ResourceVersionStatusType;
 
-  /**
-   * Based on if the resource is valid and if the final type is
-   * published it returns the text to be displayed in the publish cards
-   *
-   * @param type if it is test of prod
-   *
-   * @returns the text to display
-   */
-  const getDeploymentNotPossibleText = (type: 'test' | 'prod'): string => {
-    if (!resourceIsValid) return '';
-    if (
-      resourceIsValid &&
-      type === 'test' &&
-      newVersionText === currentEnvVersionTest &&
-      isLocalRepoInSync
-    )
-      return 'Siste versjon er allerede publisert';
-    if (
-      resourceIsValid &&
-      type === 'prod' &&
-      newVersionText === currentEnvVersionProd &&
-      isLocalRepoInSync
-    )
-      return 'Siste versjon er allerede publisert';
-    return '';
-  };
+        setVersionInTest(versions.publishedVersions.find((v) => v.environment === 'TT02').version);
+        setVersionInProd(versions.publishedVersions.find((v) => v.environment === 'PROD').version);
+        setLocalVersion(versions.resourceVersion);
+        setNewVersionText(versions.resourceVersion ?? '');
+
+        get(getValidatePolicyUrlBySelectedContextRepoAndId(selectedContext, repo, resourceId))
+          .then((validatePolicyRes) => {
+            // Remove error if status is 200
+            setHasPolicyError(validatePolicyRes.status === 200 ? 'none' : 'validationFailed');
+            setIsLoading(false);
+          })
+          .catch((err) => {
+            // If the ploicy does not exist, set it
+            if (err.response.status === 404) setHasPolicyError('notExisting');
+            setIsLoading(false);
+          });
+      })
+      .catch((err) => {
+        console.error('Error getting the publish status', err);
+        setIsLoading(false);
+        setHasError(true);
+      });
+
+    setHasResourceError(true);
+    // TODO - Validate resource when API is ready
+    /*get(getValidateResourceUrlBySelectedContextRepoAndId(selectedContext, repo, resourceId))
+      .then((res) => {
+        console.log('res', res);
+        res.status === '200' && setHasResourceError(false);
+      })
+      .catch((err) => console.log(err));
+      */
+  });
 
   /**
-   * Gets the message to be displayed in the status card
-   *
-   * @returns the message to display
+   * Constantly check the repostatus to see if we are behind or ahead of master
    */
-  const getStatusCardMessage = (): string => {
-    if (!isLocalRepoInSync) {
-      return 'Lokalt repo er ikke i sync med remote repo. Vennligst last opp og hent ned slik at du er i sync.';
-    } else {
-      if (resourceIsValid) {
-        return 'Ingen feil i ressursen.';
-      } else {
-        return 'Det er feil i ressursen - TODO - LIST ALL';
-      }
+  useEffect(() => {
+    if (repoStatus) {
+      setIsLocalRepoInSync(
+        (repoStatus.behindBy === 0 || repoStatus.behindBy === null) &&
```

Comment:
what does behindBy and aheadBy mean?

Reply:
This is Gitea's value fields for if your repo is behind or ahead of master. behindby refers to number of commits you are behind master. The value returns 0 or null when you are in sync with master.

Reply:
Allright!